### PR TITLE
Fix missing multibody loops in SolidMechanicsLagrangianFEM solver.

### PR DESCRIPTION
**Pull request by CusiniM**
These changes are important for the Multilevel EFEM-phase field work.

### CODE REVIEW

File: src/coreComponents/physicsSolvers/solidMechanics/SolidMechanicsLagrangianFEM.cpp [hunk 1]
```diff
@@ -1071,75 +1079,82 @@ SolidMechanicsLagrangianFEM::
 {
   GEOSX_MARK_FUNCTION;
 
-  MeshLevel const & mesh = domain.getMeshBody( 0 ).getMeshLevel( 0 );
-  NodeManager const & nodeManager = mesh.getNodeManager();
+  real64 totalResidualNorm = LvArray::NumericLimits< real64 >::max;
```
Comment (rrsettgast) [inline src/coreComponents/physicsSolvers/solidMechanics/SolidMechanicsLagrangianFEM.cpp:1082]: ??

**Review by rrsettgast — APPROVED** [review]

File: src/coreComponents/physicsSolvers/solidMechanics/SolidMechanicsLagrangianFEM.cpp (hunk 1 above)
Comment (CusiniM) [reply src/coreComponents/physicsSolvers/solidMechanics/SolidMechanicsLagrangianFEM.cpp:1082]: we need to do something with the residual when we have multiple bodies. I am taking the max across all mesh bodies.

File: src/coreComponents/physicsSolvers/solidMechanics/SolidMechanicsLagrangianFEM.cpp [hunk 2]
```diff
@@ -1071,75 +1079,82 @@ SolidMechanicsLagrangianFEM::
...
 
-  MpiWrapper::bcast( globalResidualNorm, 2, 0, MPI_COMM_GEOSX );
+    MpiWrapper::bcast( globalResidualNorm, 2, 0, MPI_COMM_GEOSX );
 
 
-  real64 const residual = sqrt( globalResidualNorm[0] )/(globalResidualNorm[1]+1);  // the + 1 is for the first
-                                                                                    // time-step when maxForce = 0;
+    real64 const residual = sqrt( globalResidualNorm[0] )/(globalResidualNorm[1]+1); // the + 1 is for the first
+                                                                                     // time-step when maxForce = 0;
+    totalResidualNorm = std::max( residual, totalResidualNorm );
```
Comment (CusiniM) [inline src/coreComponents/physicsSolvers/solidMechanics/SolidMechanicsLagrangianFEM.cpp:1149]: max taken here.

**Review by klevzoff — APPROVED** [review]

**Comment by CusiniM** [comment]
@rrsettgast, @klevzoff I had not noticed but, apparently, there is still one place where we assume a single `meshBody` and it's in https://github.com/GEOSX/GEOSX/blob/fa2011abd34505835fa1f0a1242a98ccfd4075dd/src/coreComponents/fieldSpecification/FieldSpecificationManager.hpp#L248. I think that, we need to change this or it won't be possible to work with multiple `meshBodies` or `meshLevels`. If I remember correctly the reason why we need this to work asap is that @andrembcosta needs to apply different b.c. in two different SolidMechanicsSolvers (basically a coarse global and a local refined one) on two different meshes but, right now, the mesh to which mechanics b.c. are applied is hardcoded to be the first one (body 0 and level 0). 

I propose that I change the` FieldspecificationManager::apply` method  so that it does not use domain but the specific `meshLevel` to which b.c. need to be applied. Any issue with that?

**Comment by rrsettgast** [comment]
> @rrsettgast, @klevzoff I had not noticed but, apparently, there is still one place where we assume a single `meshBody` and it's in
> 
> https://github.com/GEOSX/GEOSX/blob/fa2011abd34505835fa1f0a1242a98ccfd4075dd/src/coreComponents/fieldSpecification/FieldSpecificationManager.hpp#L248
> 
> I think that, we need to change this or it won't be possible to work with multiple `meshBodies` or `meshLevels`. If I remember correctly the reason why we need this to work asap is that @andrembcosta needs to apply different b.c. in two different SolidMechanicsSolvers (basically a coarse global and a local refined one) on two different meshes but, right now, the mesh to which mechanics b.c. are applied is hardcoded to be the first one (body 0 and level 0).
> I propose that I change the` FieldspecificationManager::apply` method so that it does not use domain but the specific `meshLevel` to which b.c. need to be applied. Any issue with that?

sounds good to me.

**Pull request merged**